### PR TITLE
Clarify rich display docs, tabulate reprs

### PR DESCRIPTION
**Pull request by terrdavis**
This is a draft; text wrapping is not finalized.
- when are the mimebundle and ipython_display methods available?
- can repr_pretty also return metadata?
- is the sphinx "autosectionlabel" extension enabled?
for #12992

### CODE REVIEW

**Comment by terrdavis** [comment]
I didn't try regenerating the docs. I just used GitHub's browser editor and previewed the changes to make sure they rendered properly.

File: docs/source/config/integrating.rst [hunk 1]
```diff
@@ -42,43 +87,61 @@ For example::
...
-to inform the frontend how to size the image.
 
-There are also two more powerful display methods:
+Special methods
+^^^^^^^^^^^^^^^
+
+Pretty printing
+"""""""""""""""
+
+To customize how your object is pretty-printed, add a `_repr_pretty_` method
```
Comment (Carreau) [inline docs/source/config/integrating.rst:97]:
```suggestion
To customize how your object is pretty-printed, add a ``_repr_pretty_`` method
```

**Comment by Carreau** [comment]
Thanks ! That is great !

**Comment by terrdavis** [comment]
@Carreau
I noticed the [latest docs](https://ipython.readthedocs.io/en/latest/config/integrating.html) result in a "500 internal server error".

Could this PR be the cause?
Do you know the answers to my original questions?
>* when are the mimebundle and ipython_display methods available?
>* can repr_pretty also return metadata?
>* is the sphinx "autosectionlabel" extension enabled?

**Pull request merged**